### PR TITLE
114903265 arrays homework 1 master

### PR DESCRIPTION
**Pull request by dougcass**

### CODE REVIEW

File: arrays/array_practice.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# The third element is 3
+puts "The third element is #{array[2]}"
+# The element with an index of 3 is 4
+puts "The element with an index of #{array[2]} is #{array[3]}"
+# The second from last element is 9
+puts "The second last element is #{array[8]}"
+# The first four elements are '1, 2, 3, 4'
+puts "The first four elements are '#{array.take(4).to_s.tr('[]',"")}'"
+# If we delete 5, 6 and 7 from the array, we're left with [1,2,3,4,8,9,10]
+puts "If we delete #{array.delete_at(4)}, #{array.delete_at(4)} and #{array.delete_at(4)} from the array,  we're left with #{array} "
```
Comment (timsjpark) [inline arrays/array_practice.rb]: This is a valid method, but could the code could be a little cleaner if you use `delete_if`, like so:

```
"...delete 5, 6, and 7 from the array, we're left with #{array.delete_if { # some code }"
```

File: arrays/array_practice.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+
+#Create an array variable with the the numbers 1 through 10
+array = (1..10).to_a
+# 1...2...3...4...5...6...7...8...9...10...
+puts "#{array.join('...')}" + "..."
```
Comment (jasonnoble) [inline arrays/array_practice.rb]: `puts array.join('...') + '...'`

No need for the interpolation if you have nothing else in the string...

File: arrays/array_practice.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# T-10, 9, 8, 7, 6, 5, 4, 3, 2, 1...  BLASTOFF!
+puts 'T-' + array.reverse.to_s.tr('[]',"") + '...  BLASTOFF!'
```
Comment (jasonnoble) [inline arrays/array_practice.rb]: Seems like a `join` would work better.

File: arrays/array_practice.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# The last element is 10
+puts "The last element is #{array.last}"
+# The first element is 1
+puts "The first element is #{array.first}"
+# The third element is 3
+puts "The third element is #{array[2]}"
+# The element with an index of 3 is 4
+puts "The element with an index of #{array[2]} is #{array[3]}"
```
Comment (jasonnoble) [inline arrays/array_practice.rb]: `with an index of 3 is #{array[3]}`

File: arrays/array_practice.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# The second from last element is 9
+puts "The second from last element is #{array[8]}"
```
Comment (jasonnoble) [inline arrays/array_practice.rb]: Can you find a way to find the "second from last element" no matter how many elements are in the array?  Hint: Try negative #'s

File: arrays/array_practice.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+# The first four elements are '1, 2, 3, 4'
+puts "The first four elements are '#{array.take(4).to_s.tr('[]',"")}'"
```
Comment (jasonnoble) [inline arrays/array_practice.rb]: `join` would be better than converting it to a string and `tr`ing the characters you don't want.

File: arrays/array_practice.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+# If we delete 5, 6 and 7 from the array, we're left with [1,2,3,4,8,9,10]
+#puts "If we delete #{array.delete_at(4)}, #{array.delete_at(4)} and #{array.delete_at(4)} from the array, we're left with [#{array.join(",")}] "
+puts "If we delete 5, 6 and 7 from the array, we're left with [#{array.delete_if{|array| array >4 && array < 8}.join(",")}] "
```
Comment (jasonnoble) [inline arrays/array_practice.rb:23]: Very creative.

```
array.delete(5)
array.delete(6)
array.delete(7)
puts ...
```

**Pull request merged**